### PR TITLE
fix database record duplication for updates devices/repos

### PR DESCRIPTION
**Pull request by maxamillion**
Signed-off-by: Adam Miller <admiller@redhat.com>

### CODE REVIEW

File: pkg/common/model_helpers.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
+package common
+
+import (
+	"github.com/redhatinsights/edge-api/pkg/db"
+	"github.com/redhatinsights/edge-api/pkg/models"
+	log "github.com/sirupsen/logrus"
+)
+
+// GetCommitByID
+// Pass in CommitID uint and get a *models.Commit back
```
Comment (acosferreira) [inline pkg/common/model_helpers.go:10]: func name is wrong on export, should fix it

File: pkg/common/model_helpers.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	log.Debugf("GetCommitByID::result: %#v", result)
+	log.Debugf("GetCommitByID::commit: %#v", commit)
+	if result.Error != nil {
+		return nil, result.Error
+	}
+	return &commit, nil
+}
+
+// GetCommitByOSTreeCommit
+// Pass in an OSTreeCommit string and get a *models.Commit back
```
Comment (acosferreira) [inline pkg/common/model_helpers.go:24]: func name is wrong on export, should fix it

File: pkg/common/model_helpers.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	log.Debugf("GetCommitByOSTreeCommit::commit: %#v", commit)
+	if result.Error != nil {
+		return nil, result.Error
+	}
+	return &commit, nil
+}
+
+// GetRepoByID
+// Pass in RepoID uint and get a *models.Repo back
+func GetRepoByID(repoID uint) (*models.Repo, error) {
```
Comment (acosferreira) [inline pkg/common/model_helpers.go:39]: func name is wrong on export, should fix it

File: pkg/common/model_helpers.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+	log.Debugf("GetRepoByID::result: %#v", result)
+	log.Debugf("GetRepoByID::repo: %#v", repo)
+	if result.Error != nil {
+		return nil, result.Error
+	}
+	return &repo, nil
+}
+
+// GetRepoByCommitID
+// Pass in RepoID uint and get a *models.Repo back
```
Comment (acosferreira) [inline pkg/common/model_helpers.go:52]: func name is wrong on export, should fix it

File: pkg/updates/devices.go [hunk 5]
```diff
@@ -76,9 +76,10 @@ func ReturnDevices(w http.ResponseWriter, r *http.Request) (Inventory, error) {
 func ReturnDevicesByID(w http.ResponseWriter, r *http.Request) (Inventory, error) {
 	uCtx, _ := r.Context().Value(UpdateContextKey).(UpdateContext) // this is sanitized in updates/updates
 	deviceID := uCtx.DeviceUUID
+	deviceIDParam := "&hostname_or_id=" + deviceID
```
Comment (acosferreira) [inline pkg/updates/devices.go:79]: 👍

**Review by acosferreira — APPROVED** [review]
LGTM

**Comment by maxamillion** [comment]
I've found a nil pointer dereference that's happening in a goroutine and I'm having trouble chasing it down so let's not merge just yet.
```
DEBU[2985] ::BuildUpdateRepo:uploader.UploadRepo: FINISH
DEBU[2985] ::BuildUpdateRepo:repoURL: "https://admiller-edge-api-test.s3.us-east-2.amazonaws.com/4/4/repo"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0xdda7e3]

goroutine 75 [running]:
github.com/redhatinsights/edge-api/pkg/commits.(*RepoBuilder).BuildUpdateRepo(0x1b111d8, 0xc0005668c0, 0x0, 0x1, 0x1)
        /var/home/admiller/src/dev/edge-api/pkg/commits/repobuilder.go:132 +0x723
created by github.com/redhatinsights/edge-api/pkg/updates.AddUpdate
        /var/home/admiller/src/dev/edge-api/pkg/updates/updates.go:270 +0x255
exit status 2
```

**Comment by maxamillion** [comment]
fixed, tested, and verified fixed ... this is ready for merge

File: pkg/commits/repobuilder.go [hunk 6]
```diff
@@ -128,19 +128,28 @@ func (rb *RepoBuilder) BuildUpdateRepo(ut *models.UpdateTransaction) (*models.Up
...
-		updateDone.Repo = &models.Repo{}
+	update.Status = models.UpdateStatusSuccess
+	if update.Repo == nil {
+		//  Check for the existence of a Repo that already has this commit and don't duplicate
+		var repo *models.Repo
+		repo, err = common.GetRepoByCommitID(update.CommitID)
+		if err == nil {
+			update.Repo = repo
+		} else {
+			if !(err.Error() == "record not found") {
```
Comment (biancarosa) [inline pkg/commits/repobuilder.go:139]:
```suggestion
			if err.Error() != "record not found" {
```

**Review by biancarosa — APPROVED** [review]

**Pull request merged**